providers/implementations/ciphers/cipher_aes_wrp.c

```diff
@@ -216,7 +230,7 @@ static int aes_wrap_cipher(void *vctx,
     }
 
     len = aes_wrap_cipher_internal(ctx, out, in, inl);
-    if (len == 0)
+    if (len <= 0)
         return -1;
```

Comment:
Hold on a minute...  this returns -1 through the libcrypto\<-\>provider interface.  I thought we decided that the status returns would be strictly boolean, i.e. 0 or 1 only.

As a matter of fact, returning -1 here (as well as a few lines up)  screws up this code in libcrypto:

https://github.com/openssl/openssl/blob/ed3f51ec7d2f8956bff7bedf99b65ff1595e02e8/crypto/evp/evp_lib.c#L419-L422

Reply:
Yes, this was the cause of my issue, #16387 , which inspired this pull request. Glad to see this being fixed.